libs/message-coder/src/types.ts

```diff
@@ -0,0 +1,153 @@
+import { isResult, ok, Result } from '@votingworks/types';
+import { Buffer } from 'buffer';
+
+/**
+ * Errors that can occur when encoding or decoding.
+ */
+export enum CoderError {
+  SmallBuffer = 'SmallBuffer',
+  InvalidValue = 'InvalidValue',
+  UnsupportedOffset = 'UnsupportedOffset',
+}
+
+/**
+ * Descriptive alias for `number`.
+ */
+export type BitOffset = number;
+
+/**
+ * Descriptive alias for `number`.
+ */
+export type ByteOffset = number;
+
+/**
+ * Descriptive alias for `number`.
+ */
+export type BitLength = number;
+
+/**
+ * Descriptive alias for `number`.
+ */
+export type ByteLength = number;
+
+/**
+ * Descriptive alias for `number`.
+ */
+export type Uint2 = number;
```

Comment:
Why did you not list the possible values for the smaller types, as you did in the `ballot-encoder` libs `bits` helpers?

Not listing them is great as far as I'm concerned because it avoids type assertions that convolute the code.

Reply:
Mostly I felt they were not adding to the safety of the code much and were cumbersome to work with, so I just made them all `number`.

Reply:
I think Typescript lacks the ability to narrow such a type effectively. For example, `export type UInt1 = 0 | 1`. If I filter on `x >= 0 && x < 2` the linter still won't accept it as `UInt1`. But if typescript ever gets there, may be worth it!